lib/models/room.js

```diff
@@ -401,6 +410,27 @@ Room.prototype.addReceipt = function(event) {
     });
 };
 
+/**
+ * Update the room-tag event for the room.  The previous one is overwritten.
+ * @param {MatrixEvent} event the m.tag event
+ */
+Room.prototype.addTags = function(event) {
+    // event content looks like:
+    // content: {
+    //    tags: {
+    //       $tagname: { $metadata: $value },
+    //       $tagname: { $metadata: $value },
+    //    }
+    // }
+
+    // XXX: we could do a deep-comparison to see if the tags have really
+    // changed - but do we want to bother?
+    this.emit("Room.tags", this);
+
+    // XXX: do we need to deep copy here?
+    this._tags = event.content.tags;
```

Comment:
The rest of the SDK tends to persist the *event* as a private variable which is then yanked as it needs to be accessed. This makes it clearer that the tags are being drawn from the event and not copied.

As an aside, `event` in this case is a `MatrixEvent` which means `event.content === undefined`. Use `event.getContent()` instead.